include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -344,9 +694,484 @@ class alignment_file_format_sam
+
+        switch (type_id)
+        {
+            case 'A' : // char
+            {
+                target[tag] = static_cast<char>(*std::ranges::begin(stream_view));
+                std::ranges::next(std::ranges::begin(stream_view)); // skip char that has been read
+                break;
+            }
+            case 'i' : // int32_t
+            {
+                int32_t tmp;
+                read_field(stream_view, tmp);
+                target[tag] = tmp;
+                break;
+            }
+            case 'f' : // float
+            {
+                float tmp;
+                read_field(stream_view, tmp);
+                target[tag] = tmp;
+                break;
+            }
+            case 'Z' : // string
+            {
+                target[tag] = std::string(stream_view);
+                break;
+            }
+            case 'H' :
+            {
+                // TODO
+                break;
+            }
+            case 'B' : // Array. Value type depends on second char [cCsSiIf]
+            {
+                char array_value_type_id = *std::ranges::begin(stream_view);
+                std::ranges::next(std::ranges::begin(stream_view)); // skip char read before
+                std::ranges::next(std::ranges::begin(stream_view)); // skip first ','
+
+                switch (array_value_type_id)
+                {
+                    case 'c' : // int8_t
+                    {
+                        read_sam_dict_vector(target[tag], stream_view, int8_t{});
+                        break;
+                    }
+                    case 'C' : // uint8_t
+                    {
+                        read_sam_dict_vector(target[tag], stream_view, uint8_t{});
+                        break;
+                    }
+                    case 's' : // int16_t
+                    {
+                        read_sam_dict_vector(target[tag], stream_view, int16_t{});
+                        break;
+                    }
+                    case 'S' : // uint16_t
+                    {
+                        read_sam_dict_vector(target[tag], stream_view, uint16_t{});
+                        break;
+                    }
+                    case 'i' : // int32_t
+                    {
+                        read_sam_dict_vector(target[tag], stream_view, int32_t{});
+                        break;
+                    }
+                    case 'I' : // uint32_t
+                    {
+                        read_sam_dict_vector(target[tag], stream_view, uint32_t{});
+                        break;
+                    }
+                    case 'f' : // float
+                    {
+                        read_sam_dict_vector(target[tag], stream_view, float{});
+                        break;
+                    }
+                    default:
+                        throw format_error{std::string("[SAM Parse Error] The first character in the numerical ") +
```

Comment:
for some exceptions you give the prefix "[SAM Parse Error]" but for others you don't....
also please {} initialise the string.